internal/featuretests/v3/cluster_test.go

```diff
@@ -623,9 +624,18 @@ func TestUnreferencedService(t *testing.T) {
 	rh, c, done := setup(t)
 	defer done()
 
-	rh.OnAdd(fixture.NewService("kuard").
-		WithPorts(v1.ServicePort{Port: 80, TargetPort: intstr.FromString("8080")}),
-	)
+	// Equals(...) only checks resources, so explicitly
+	// check version & nonce here and subsequently.
+	assertEqualVersion := func(t *testing.T, expected string, r *Response) {
+		t.Helper()
+		assert.Equal(t, expected, r.VersionInfo, "got unexpected VersionInfo")
+		assert.Equal(t, expected, r.Nonce, "got unexpected Nonce")
+	}
```

Comment:
Not strictly needed for this PR, but it would be nice to move this to be a method on the `Response` struct, so you could do `r.EqualVersion("1")` similarly to how you can do `r.Equals(...)` currently. That would replace the `assertEqualVersion` helper here and in `secrets_test.go`.